rust/lance/src/utils/tokio.rs

```diff
@@ -0,0 +1,19 @@
+use crate::Result;
+
+use futures::{Future, FutureExt};
+
+/// Spawn a CPU intensive task
+///
+/// This task will be put onto the rayon thread pool dedicated for CPU-intensive work
+pub fn spawn_cpu<F: FnOnce() -> Result<R> + Send + 'static, R: Send + 'static>(
+    func: F,
+) -> impl Future<Output = Result<R>> {
+    let (send, recv) = tokio::sync::oneshot::channel();
+    rayon::spawn(|| {
```

Comment:
does rayon limit parallelism in anyway? Could we add an envvar to limit the parallelism?

Reply:
It defaults to one thread per CPU.  You can use RAYON_NUM_THREADS to adjust.